store/tikv/rawkv.go

```diff
@@ -298,3 +322,106 @@ func (c *RawKVClient) sendDeleteRangeReq(startKey []byte, endKey []byte) (*tikvr
 		return resp, actualEndKey, nil
 	}
 }
+
+func (c *RawKVClient) sendBatchPut(bo *Backoffer, keys, values [][]byte) error {
```

Comment:
Why pass backoffer from parameter?

Reply:
One batch put will be split into several subtasks which will be done parallelly, so we should use the common ancestor `backoff` to control the task.

The `sendBatchPut` is invoked by `BatchPut` and `doBatchPut`.

Reply:
Could you avoid recursion?

Reply:
The `regionErr` means that the region is right, we should resplit the keys and send the requests.
The above operations are the same as the `sendBatchPut`, so I call the function recursively.
Now I can't avoid the recursion.